replica/table.cc

```diff
@@ -253,7 +243,7 @@ table::make_reader_v2(schema_ptr s,
     });
 
     const auto bypass_cache = slice.options.contains(query::partition_slice::option::bypass_cache);
-    if (cache_enabled() && !bypass_cache && !(reversed && _config.reversed_reads_auto_bypass_cache())) {
+    if (cache_enabled() && !bypass_cache && !(slice.is_reversed() && _config.reversed_reads_auto_bypass_cache())) {
```

Comment:
I wonder why we have reversed_reads_auto_bypass_cache (maybe due to earlier cache weakness in this area?). We can consider removing it too.

Reply:
We had native reverse implementation for sstables, before we had it for cache. So we had this option to bypass the cache, which at the time would still use the old reverse-whole-partition approach.